tooling/miniutil/src/build/ty.rs

```diff
@@ -12,10 +12,28 @@ pub fn ref_ty(pointee: PointeeInfo) -> Type {
     Type::Ptr(PtrType::Ref { mutbl: Mutability::Immutable, pointee })
 }
 
+pub fn ref_ty_for(ty: Type) -> Type {
+    ref_ty(PointeeInfo {
+        layout: ty.layout::<x86_64>(),
+        inhabited: true,
+        freeze: true,
+        unpin: true,
+    })
+}
+
 pub fn ref_mut_ty(pointee: PointeeInfo) -> Type {
     Type::Ptr(PtrType::Ref { mutbl: Mutability::Mutable, pointee })
 }
 
+pub fn ref_mut_ty_for(ty: Type) -> Type {
+    ref_mut_ty(PointeeInfo {
+        layout: ty.layout::<x86_64>(),
```

Comment:
```suggestion
        layout: ty.layout::<DefaultTarget>(),
```
Applies everywhere else as well.